### PR TITLE
661 period 0

### PR DESCRIPTION
**Pull request by AntonZogk**
# 661 period 0

# Summary

Adding a new feature, appending back data to staged data, these are essential for applying the correct imputation method 
# Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe):

# Checklists

This pull request meets the following requirements:

## Creator Checklist

- [x] Installable with all dependencies recorded
- [x] Runs without error
- [x] Follows PEP8 and project-specific conventions
- [x] Appropriate use of comments, for example, no descriptive comments
- [x] Functions documented using Numpy style docstrings
- [x] Assumptions and decisions log considered and updated if appropriate
- [x] Unit tests have been updated to cover essential functionality for a reasonable range of inputs and conditions
- [x] Other forms of testing such as end-to-end and user-interface testing have been considered and updated as required

If you feel some of these conditions do not apply for this pull request, please
add a comment to explain why.

## Reviewer Checklist

- [ ] Test suite passes (locally as a minimum)
- [ ] Peer reviewed with review recorded

# Additional Information

Please provide any additional information or context that would help the reviewer understand the changes in this pull request.

# Related Issues

Link any related issues or pull requests here.

### CODE REVIEW

File: mbs_results/config.json [hunk 1]
```diff
@@ -129,6 +135,20 @@
 
     "census_extra_calibration_group": [5043, 5113, 5123, 5203, 5233,
      5403, 5643, 5763, 5783, 5903, 6073],
-     "filter_out_questions": [11, 12 ,146]
+     "filter_out_questions": [11, 12 ,146],
+     "csw_to_spp_columns":{
```
Comment (Jday7879) [inline mbs_results/config.json:139]: Has this been added as the back data is from CSW?

File: mbs_results/staging/back_data.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    Applies checks on period 0 (back data period)
+
+    Staged data need back data (or period 0) to succesfully impute values,
+    this function ensures the date in back data is correctly configured.
+
+    Parameters
+    ----------
+    back_data_period : pd.Series
+        All dates in back data.
+    first_period : pd.Datetime
```
Comment (Jday7879) [inline mbs_results/staging/back_data.py:25]: first period is pandas dataframe, is this correct. function def says timestamp :)

File: mbs_results/staging/back_data.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+    return True
+
+
+def read_back_data(config: dict) -> pd.DataFrame:
+    """Loads all required files for back data as a dataframe.
+
+    Parameters
+    ----------
+    config : dict
```
Comment (Jday7879) [inline mbs_results/staging/back_data.py:69]: I like this way of showing what variables are loaded from the config

File: mbs_results/staging/back_data.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+    imp_marker_col = config["imputation_marker_col"]
+
+    back_data = read_back_data(config)
+
+    back_data = back_data.rename(columns=config["csw_to_spp_columns"], errors="raise")
+
+    # Json file can't store keys as int, thus stored them as str
+    # This is why we need to convert them to str here since from csv source
+    # they are loaded as int
```
Comment (Jday7879) [inline mbs_results/staging/back_data.py:136]: Makes sense to me, nice comments

**Review by Jday7879 — APPROVED** [review]
Really good work here Anton! Couple of minor comments but looks good to me for the mot part.

File: mbs_results/config.json (hunk 1 above)
Comment (AntonZogk) [reply mbs_results/config.json:139]: You mean `csw_to_spp_columns`  dict?

File: mbs_results/staging/back_data.py (hunk 2 above)
Comment (AntonZogk) [reply mbs_results/staging/back_data.py:25]: good spot !

**Comment by Jday7879** [comment]
@AntonZogk Doc changes look good to me, happy for it to be merged whenever you are :)

File: mbs_results/config.json (hunk 1 above)
Comment (AntonZogk) [reply mbs_results/config.json:139]: There is a mismatch with column names with csw data and spp data for an example csw has question_no while spp has questioncode , so this converts the csw column to match the spp ones, for now just these 3 need renaming

**Pull request merged**